### PR TITLE
Fix `listing` block in SSR, now it is fully variations aware, and the…

### PR DESCRIPTION
**Pull request by sneridagh**
… config is passed to the SSR querystring action

### CODE REVIEW

**Review by tiberiuichim — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### Bugfix
 
+- Fix `listing` block in SSR, now it's fully variations aware, and the config is passed to the SSR querystring action @sneridagh
```
Comment (stevepiercy) [inline CHANGELOG.md:11]:
```suggestion
- Fix `listing` block in SSR, now that it is fully variations aware and the configuration is passed to the SSR `querystring` action. @sneridagh
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Review by stevepiercy — APPROVED** [review]

**Comment by Rudd-O** [comment]
This seems correct in my view, though I have no way to effectively evaluate it.  Tests seem to pass.

The fact that the promise of rendering a site to nonjs visitors / search engines was broken... and this seems to be the direction that the stewards of Plone are going with... breaks my heart.

It is the obvious unspoken conclusion that a client-side rendering strategy was eventually going to get us all to the point that client rendering (= js) turned off will net visitors no (or broken) client rendering, no matter how much we want to render for the client server-side.  We just can't test for that on every individual Web site.  If the thing renders the titles but won't render the contents, and there's the odd idiot like me who browses the web without cookies or JS by default, well who's gonna test that?  RMS is busy reading email from 1989.

This should probably be merged anyway.

Sad.

**Review by Rudd-O — COMMENTED** [review]
^^^

**Comment by davisagli** [comment]
@Rudd-O Your response to someone fixing a server-side rendering issue is to criticize the team for not caring enough about server-side rendering? Wow, I bet that doesn't help motivate the people that are volunteering time to work on it.

I think there are feasible strategies for testing that server-side rendering isn't broken. i.e. load key pages as an anonymous user with cypress and test that the DOM is the same before and after client-side rendering. If you care about it, it'd be more effective to work on making it better than complain about the current situation not meeting your needs.

I hope this doesn't come across as disrespectful. To some extent, I can understand your frustration. But your comment struck a bit of a nerve because I know the Volto team cares about and has put effort into this.

**Comment by stevepiercy** [comment]
https://2022.ploneconf.org looks great to me!

Screen Shot 2022-09-26 at 6 48 29 PM

**Comment by davisagli** [comment]
@stevepiercy The listing of talks is missing key text though (speaker and abstract). I think that is what this will help with...

**Comment by tiberiuichim** [comment]
@Rudd-O I want to point out that it's not "Volto developers don't care about JS disabled", as @sneridagh and I worked hard to make that happen. It's a bug due to a rather new / uncommon feature that was added to Volto. All this is rather fresh stuff, probably around 6 months. There's a ton of development to do in Volto world, we're trying to make a really attractive platform for developers.

**Review by mpeeters — APPROVED** [review]

**Comment by sneridagh** [comment]
@Rudd-O Thanks for report this bug on Discord, good catch! 

First of all, I have to state in case of someone reach this PR at some point, and reads it, that Volto in SSR rendering works well, as the many projects developed by the community that are currently in production since _years_ ago. 

This piece of code is probably one of the more complex in Volto, as usually happens when talking about SSR features. It's one that @tiberiuichim and myself spent quite a few time to figure it out and find the right implementation. Finally we come up with a solution inspired in how NextJS is doing things and we manage to address it. Yes, it had an issue, you identified it and pointed it out in the Discord channel and it got fixed within the day... This says a lot about this community, right?

It's true that this community still haven't found time to test and assess these edge cases (since plain page SSR works since years ago), but we will eventually do. Maybe you could take a look and help with it?

I wish we could do something to cheer you up. Volto is not the default UI for Plone 6 because "the stewards of Plone" decided so. It's because the Plone community decided so. Plone 6 Volto is a proven solution that powers dozens of websites nowadays and renew its prowess every day. Are you attending the conference? I'd like to be able to address and convince you of these points and dissipate any concern you could have about the future of Plone.

**Comment by Rudd-O** [comment]
> @Rudd-O Your response to someone fixing a server-side rendering issue is to criticize the team for not caring enough about server-side rendering? Wow, I bet that doesn't help motivate the people that are volunteering time to work on it.
> 
> I think there are feasible strategies for testing that server-side rendering isn't broken. i.e. load key pages as an anonymous user with cypress and test that the DOM is the same before and after client-side rendering. If you care about it, it'd be more effective to work on making it better than complain about the current situation not meeting your needs.
> 
> I hope this doesn't come across as disrespectful. To some extent, I can understand your frustration. But your comment struck a bit of a nerve because I know the Volto team cares about and has put effort into this.

No disrespect taken.  I am frustrated, but I recognize that the Volto team puts effort in ensuring that both alternatives work fine.

Totally stoked, however, at how quickly the issue was remedied.  Yes, that does say a lot about the community, and it's only positive things.  Let's figure out a way to never have this issue crop up in the future ever again.
